Fix socket service API usage with native_sim

jukkar opened the pull request:
Add build checks that try to make sure that we are not trying to compile socket service API that uses eventfd, with host libc as that will not work. Because of this change, my earlier bad fix for this is reverted.
The last three commits are also part of PR #73212

jukkar committed:
Revert "net: sockets: Check eventfd errors better"

This reverts commit b56297bf161e739cd53eb7783f1e0bd5d8ead2b1
because it is causing problems when using native_sim and not
having CONFIG_POSIX_API set.

The proper fix is to set CONFIG_POSIX_API when using eventfd
and native_sim so that picolibc is used instead of host libc.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: dns: Fix resolver debug print

Make sure that we are not trying to print NULL pointer string.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
tests: net: dns_resolve: Increase the number of socket descriptors

The test needs couple of more socket/file descriptors to work because
CONFIG_POSIX_API creates three more file descriptors namely the
stdin, stdout and stderr descriptors.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: sockets_service: Add build check for using zephyr libc

Add checks to make sure that we are not trying to use the socket
service library with eventfd if CONFIG_POSIX_API is not set and if
using native_sim based board. The reason is that we should always
use zephyr libc based eventfd implementation instead of host libc one.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

SeppoTakalo reviewed (dismissed):
This fixes the LwM2M client on `native_sim`

jukkar committed:
shell: backend: telnet: Rename write and read functions

Rename write() to telnet_write(), and read() to telnet_read() so
that if we enable CONFIG_POSIX_API, the compiler will not complain
about those two functions as they conflict with POSIX API ones.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar commented:
* CI fixes
* First 6 commits are from #73212 in order the CI to pass without errors.

rlubos reviewed (dismissed):
LGTM
